app/lib/ingestors/google/gmail.rb

```diff
@@ -76,6 +77,17 @@ def extract_headers(headers)
     headers.each_with_object({}) { |header, hash| hash[header.name.downcase] = header.value }
   end
 
+  def search_query
+    latest_received_at = user.documents
+                             .joins("INNER JOIN gmail_messages ON documents.documentable_id = gmail_messages.id AND documents.documentable_type = 'GmailMessage'")
+                             .maximum('gmail_messages.received_at')
```

Comment:
Another idea we could so that it generalises for all the ingestors would could be to store sync_status in a separate table. That way we could track last_sync_started_at, last_sync_ended_at, latest_document_sync_time for all ingestors.

Wdyt?